### DISCUSSION TITLE
Strukturierung von JSON Daten

### DISCUSSION THREAD

**Discussion by AndreHaupt85** [neuefische/questions#165] [Web] [answered]
Hallo,

ich versuche grade eine Detailpage aus json Daten zu rendern. 
Leider scheitert es schon am Aufbau der JSON Daten da ich je nach Objekt eine unterschiedliche Anzahl von informationen habe.

Ich weiß nicht wie sinnvoll es ist Arrays und Objeke zu verschachteln und wenn ja wie oft und tief man das tun sollte.

Könnte mir dabei jemand Helfen?

![Bildschirmfoto 2022-09-23 um 14 51 13](https://user-images.githubusercontent.com/109967968/191964165-5b705ade-66aa-4660-84db-51f824617333.png)

**Comment by NikRoe** [comment, marked as answer]
Hallo Andre,

das sieht mir auf den ersten Blick nach einem Rezept für einen Cocktail aus und das würde ich möglichst simpel aufbauen. Wie wäre es hiermit:
```json
[
  {
    "id": "R-000001",
    "name": "Name des Cocktails",
    "ingredients": [
      { "name": "Tequila", "amount": "3cl" },
      { "name": "Granadine", "amount": "1cl" }
    ],
    "instructions": "Shake it and drink it",
    "type": "alkoholhaltig",
    "category": "tropical"
  }
]
```

die weiteren Cocktails würde ich in einer ähnlichen Struktur aufbauen.

Hilft dir das schon weiter? 🤞
